### DISCUSSION TITLE
Adding plugins does not participate in packaging

### DISCUSSION THREAD

**Discussion by dongchen1221** [eclipse-tycho/tycho#2393] [Q&A]
A single plugin can generate the correct zip package, but when a new plugin is added, its content always does not exist in the zip generated.

The parent pom file is as follows, with the same structure as ilg.gnumcueclipse.debug.gdbjtag.custom and com.nucleisys.ide. But currently, only zip packages containing com.nucleisys.ide can be generated

  <modules>
    <module>com.nucleisys.ide.target-platform</module>
    <module>com.nucleisys.ide</module>
    <module>com.nucleisys.ide.feature</module>
    <module>ilg.gnumcueclipse.debug.gdbjtag.custom</module>
    <module>ilg.gnumcueclipse.debug.gdbjtag.custom.feature</module>
    <module>com.nucleisys.ide.repository</module>
  </modules>
   <properties>
    <ide.version>1.1.6</ide.version>  
    <tycho.version>2.7.5</tycho.version>
    <tycho.scmUrl>scm:git:https://git.eclipse.org/r/p/egit/egit.git</tycho.scmUrl>
  </properties>
  
  	<pluginRepositories>
		<pluginRepository>
			
			<id>cbi</id>
			<url>https://repo.eclipse.org/content/repositories/cbi-releases/</url>
		</pluginRepository>
	</pluginRepositories>
	
   <repositories>
   <repository>
     <id>eclipse</id>
     <layout>p2</layout>
     <url>https://download.eclipse.org/releases/2022-06</url>
   </repository>
  </repositories>
  
  <build>
    <plugins>
      <plugin>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-maven-plugin</artifactId>
        <version>${tycho.version}</version>
        <extensions>true</extensions>
      </plugin>
      
      <plugin>
			<groupId>org.eclipse.tycho</groupId>
			<artifactId>tycho-compiler-plugin</artifactId>
			<version>${tycho.version}</version>
			<configuration>
				<encoding>UTF-8</encoding>
				
				
			</configuration>
		</plugin>
      
      <plugin>
		<groupId>org.eclipse.tycho</groupId>
		<artifactId>target-platform-configuration</artifactId>
		<version>${tycho.version}</version>
		<configuration>
			<resolver>p2</resolver>
			<pomDependencies>consider</pomDependencies>
			<environments>
				<environment>
					<os>linux</os>
					<ws>gtk</ws>
					<arch>x86_64</arch>
				</environment>
				<environment>
					<os>linux</os>
					<ws>gtk</ws>
					<arch>aarch64</arch>
				</environment>
				<environment>
					<os>win32</os>
					<ws>win32</ws>
					<arch>x86_64</arch>
				</environment>
			</environments>
			<target>
				<artifact>
					  <groupId>com.nucleisys</groupId>
					  <artifactId>com.nucleisys.ide.target-platform</artifactId>
					  <version>1.1.6</version>
				</artifact>
			</target>
		</configuration>
	</plugin>
			
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>tycho-source-plugin</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<distinctSourceRoots>false</distinctSourceRoots>
					<strictSrcIncludes>false</strictSrcIncludes>
					<archive>
						<addMavenDescriptor>false</addMavenDescriptor>
					</archive>
				</configuration>
				<executions>
					<execution>
						<id>attach-source</id>
						<goals>
							<goal>plugin-source</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
			
			<plugin>
				<groupId>org.eclipse.tycho.extras</groupId>
				<artifactId>tycho-source-feature-plugin</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<archive>
						<addMavenDescriptor>true</addMavenDescriptor>
					</archive>
					<reuseBrandingPlugin>false</reuseBrandingPlugin>
				</configuration>
				<executions>
					<execution>
						<id>source-feature</id>
						<phase>package</phase>
						<goals>
							<goal>source-feature</goal>
						</goals>
						<configuration>
							<excludes>
								<plugin id="com.nucleisys.ide.feature" />
								<plugin id="ilg.gnumcueclipse.debug.gdbjtag.custom.feature" />
							</excludes>
						</configuration>
					</execution>
				</executions>
			</plugin>
			
			
			<plugin>
	          <groupId>org.eclipse.tycho</groupId>
	          <artifactId>tycho-p2-repository-plugin</artifactId>
	          <version>${tycho.version}</version>
        	</plugin>	
			
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>tycho-p2-plugin</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<generateDownloadStatsProperty>true</generateDownloadStatsProperty>
				</configuration>
				<executions>
					<execution>
						<id>attached-p2-metadata</id>
						<phase>package</phase>
						<goals>
							<goal>p2-metadata</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
    </plugins>
  </build>
  <profiles>
		
		<profile>
			<id>feature-no-source</id>
			<activation>
				<file>
				<exists>feature-no-source.properties</exists>
				</file>
			</activation>
			<build>
				<plugins>
					<plugin>
						<groupId>org.eclipse.tycho.extras</groupId>
						<artifactId>tycho-source-feature-plugin</artifactId>
						<version>${tycho.version}</version>
						<executions>
							<execution>
								<id>source-feature</id>
								<phase>none</phase>
							</execution>
						</executions>
					</plugin>
				</plugins>
			</build>
		</profile>
		<profile>
			<id>production</id>
			<build>
				<plugins>
					<plugin>
						<groupId>org.eclipse.cbi.maven.plugins</groupId>
						<artifactId>eclipse-jarsigner-plugin</artifactId>
						<version>${cbi-plugins.version}</version>
						<executions>
							<execution>
								<id>sign</id>
								<phase>verify</phase>
								<goals>
									<goal>sign</goal>
								</goals>
							</execution>
						</executions>
					</plugin>
				</plugins>
			</build>
		</profile>
	</profiles>
</project>

**Comment by laeubi** [comment]
> A single plugin can generate the correct zip package, but when a new plugin is added, its content always does not exist in the zip generated.

Its not completely sure what you mean, maybe you have a small exapmple that shows the problem? In generla, a new plugin must be referenced somewhere (e.g. in a product a feature or as a dependency of an existing plugin) to be included in a product/update-site zip.
